### DISCUSSION TITLE
Use sphinx-hoverxref in traceability links

### DISCUSSION THREAD

**Discussion by David-Le-Nir** [useblocks/sphinx-needs#1348] [General]
I'd like to use [sphinx-hoverxref](https://sphinx-hoverxref.readthedocs.io/en/latest/index.html) on traceability links of my needs.

For example, if I have something like 

```
.. req:: my req
   :link: other-external-req
```

I'd like to have the hoverxrek applying on the link (i.e. when I hover the _other-external-req_ in html see the tooltip with the content). The main issue is that we cannot use a role in a directive option.

I tried multiple approaches, obviously not working, such as

with role on link option
```
.. req:: my req
   :link: :hoverxref:`other-external-req`
```

with role on an option not transformed to html link
```
.. req:: my req
   :link: other-external-req
   :a_non_link_field: :hoverxref:`other-external-req <url ...>`
```

with a dynamic function returning the string :hoverxref:`other-external-req <url ...>` or the raw html
```
.. req:: my req
   :link: other-external-req
   :a_non_link_field: [[a_dynamic_function('other-external-req')]]
```

with the raw html
```
.. req:: my req
   :link: other-external-req
   :a_non_link_field: <a class="hxr-hoverxref hxr-modal reference external" href="http://...">other-external-req</a>
```

Do you think it is achievable ? If so, what would you try ?

The extension is working well in the need content but this is not what I want ^^

thanks in advance

**Comment by PhilipPartsch** [comment]
It is currently not possible to write rst syntax in needs options or links.
To make it possible is a lot of effort.

But you could try the extension [sphinx-preview](https://github.com/useblocks/sphinx-preview).

**Reply by David-Le-Nir** [reply]
Thanks a lot, it seems exactly what I want to achieve
